### ISSUE TITLE
DetailsList check-all checkbox doesn't have a name property nor the option to add an aria-label

### ISSUE THREAD

**Issue by satbai** [OfficeDev/office-ui-fabric-react#3486]
### Bug Report
- __Package version(s)__: latest

#### Priorities and help requested (not applicable if asking question):
Are you willing to submit a PR to fix? No
Requested priority: (Blocking, High, Normal, Low) Normal

#### Describe the issue:
The DetailsList check-all checkbox should have a name property, and/or it should have the option for a custom aria-label to meet the MAS C accessibility standards. 

User impact: Visually impaired users, who rely on screen reader will not be able to know the name of the check box.

#### Actual behavior:
Screen shot taken from https://developer.microsoft.com/en-us/fabric#/components/detailslist 
![image](https://user-images.githubusercontent.com/24945574/33444292-b95bb144-d5ae-11e7-96e5-948c59b31387.png)

#### Expected behavior:
The checkbox should have a name property, e.g. "check/uncheck all".

**Comment by erichdev** [comment]
DetailsList has the prop `ariaLabelForSelectAllCheckbox` so you can set a label for that checkbox. The second example on the docs site uses that prop if you'd like to test and inspect. 

Closing issue, please reopen if any issues remain 👍
